internal/pkg/runtime/engine/apptainer/prepare_linux.go

```diff
@@ -1521,12 +1521,14 @@ func (e *EngineOperations) setUserInfo(useTargetIDs bool) {
 			e.EngineConfig.JSON.UserInfo.Gecos = pw.Gecos
 			e.EngineConfig.JSON.UserInfo.UID = int(pw.UID)
 			e.EngineConfig.JSON.UserInfo.GID = int(pw.GID)
+			e.EngineConfig.JSON.UserInfo.Shell = pw.Shell
 		}
 	} else {
 		e.EngineConfig.JSON.UserInfo.Username = pw.Name
 		e.EngineConfig.JSON.UserInfo.Gecos = pw.Gecos
 		e.EngineConfig.JSON.UserInfo.UID = int(pw.UID)
 		e.EngineConfig.JSON.UserInfo.GID = int(pw.GID)
+		e.EngineConfig.JSON.UserInfo.Shell = pw.Shell
```

Comment:
While you're in there, can't these 5 lines of code all be under one case of `if err == nil` ?  I mean,
```
if useTargetIDs {
		pw, err = user.GetPwUID(uint32(e.EngineConfig.GetTargetUID()))
}
if err == nil {
		... 5 lines ...
}
```

Reply:
Done